### PR TITLE
Add the ability to remove request port or explicitly set it per the URL scheme default.

### PR DESCRIPTION
**Pull request by cpasillas**

### CODE REVIEW

File: port/port_modifier.go [hunk 1]
```diff
@@ -16,47 +16,73 @@ package port
...
-	return &Modifier{
-		port: port,
-	}
+// NewModifier returns a RequestModifier that can be configured to alter the request URL and Host header's port.
+// One of DefaultPortForScheme, UsePort, or RemovePort should be called to configure this modifier.
+func NewModifier() *Modifier {
+	return &Modifier{}
+}
+
+// Explicitly specify 80 for HTTP or 443 for HTTPS ('http://example.com:80'). Do nothing for a scheme that is not 'http' or 'https'.
```
Comment (bramhaghosh) [inline port/port_modifier.go]: The first sentence should be a one-sentence summary that starts with the name being declared.

https://golang.org/doc/effective_go.html#commentary

File: port/port_modifier.go [hunk 2]
```diff
@@ -16,47 +16,73 @@ package port
...
+// This overrides any previous configuration for this modifier.
+func (m *Modifier) DefaultPortForScheme() {
+	m.defaultForScheme = true
+	m.remove = false
+}
+
+// Explicitly add the specified port to the host string ('example.com:1234').
```
Comment (bramhaghosh) [inline port/port_modifier.go]: The first sentence should be a one-sentence summary that starts with the name being declared.

https://golang.org/doc/effective_go.html#commentary

File: port/port_modifier.go [hunk 3]
```diff
@@ -16,47 +16,73 @@ package port
...
+	m.port = port
+	m.remove = false
+	m.defaultForScheme = false
 }
 
-// ModifyRequest alters the request URL and Host header to explictly
-// use the provided port.  In the case that the provided port is
-// the default port for the schema, the port will be explictly
-// declared.
+// Remove the port from the host string ('example.com').
```
Comment (bramhaghosh) [inline port/port_modifier.go]: The first sentence should be a one-sentence summary that starts with the name being declared.

https://golang.org/doc/effective_go.html#commentary

File: port/port_modifier_test.go [hunk 4]
```diff
@@ -44,6 +45,94 @@ func TestPortModifierOnRequest(t *testing.T) {
 	}
 }
 
+// Test that a modifier with no configuration does nothing.
```
Comment (bramhaghosh) [inline port/port_modifier_test.go]: no need for comment

File: port/port_modifier.go (hunk 1 above)
Comment (cpasillas) [inline port/port_modifier.go]: Done.

File: port/port_modifier.go (hunk 2 above)
Comment (cpasillas) [inline port/port_modifier.go]: Done.

File: port/port_modifier.go (hunk 3 above)
Comment (cpasillas) [inline port/port_modifier.go]: Done.

File: port/port_modifier_test.go (hunk 4 above)
Comment (cpasillas) [inline port/port_modifier_test.go]: Done.

**Pull request merged**